connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -199,6 +213,11 @@ protected static ConfigDef baseConfigDef() {
                         null,
                         Importance.LOW,
                         PLUGIN_PATH_DOC)
+                .define(PLUGIN_DISCOVERY_CONFIG,
+                        Type.STRING,
+                        PluginDiscoveryMode.HYBRID_WARN.toString(),
```

Comment:
We can add a validator for this:
```suggestion
                        PluginDiscoveryMode.HYBRID_WARN.toString(),
                        ConfigDef.CaseInsensitiveValidString.in(Utils.enumOptions(PluginDiscoveryMode.class)),
```

It may not be applied before we parse the property value, but it also helps with our autogenerated docs for the property (for example, see the "Valid Values" section in the [docs for exactly.once.source.support](https://kafka.apache.org/documentation.html#connectconfigs_exactly.once.source.support)).